### PR TITLE
[WIP] cluster: Secure cluster traffic via mutual TLS

### PR DESCRIPTION
**Pull request by mxinden**
Use github.com/mxinden/memberlist-tls-transport to optionally secure the
Alertmanager cluster traffic via mutual tls.

This is an early work-in-progress version, but one can already test it out locally. I added everything to get up and running quickly:

1. `go get -u github.com/cloudflare/cfssl/cmd/...`
2. `cd examples/ha/tls/ &&  make start`

### CODE REVIEW

File: cluster/cluster.go [hunk 1]
```diff
@@ -210,6 +219,46 @@ func Create(
 		p.setInitialFailed(resolvedPeers, bindAddr)
 	}
 
+	// TODO: Don't just dereference.
```
Comment (brian-brazil) [inline cluster/cluster.go]: We're handling all this differently in the node exporter tls stuff, which you should probably vendor/hack in here to keep things in line

Comment (mxinden) [reply cluster/cluster.go]: For sure. I will align this once https://github.com/prometheus/node_exporter/pull/1277 is merged.

**Comment by hooten** [comment]
@mxinden Do you plan to pick this back up? If not, I would like to. Thanks!

**Comment by mxinden** [comment]
@hooten I have no plans to work on this, but I am more than happy to answer any questions in case you would like to continue the effort. Would be great to see this in `master` at some point.

As far as I am aware this is still aligned with the Alertmanager roadmap. I am //CCing @simonpasquier here just to make sure.

**Comment by sharadgaur** [comment]
@hooten Love to pair with you on this :)

**Comment by mxinden** [comment]
With https://github.com/prometheus/alertmanager/pull/2237 merged, I am closing here.

**Pull request closed without merging**